sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v2/script.sql

```diff
@@ -0,0 +1,415 @@
+      FROM
+        stub_session_ids_staging
+      GROUP BY
+        ga_client_id,
+        ga_session_id
+    ),
+    landing_page_by_session_staging AS (
+      SELECT
+        user_pseudo_id AS ga_client_id,
+        CAST(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'ga_session_id'
+            LIMIT
+              1
+          ).int_value AS string
+        ) AS ga_session_id,
+        SPLIT(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'page_location'
+            LIMIT
+              1
+          ).string_value,
+          '?'
+        )[OFFSET(0)] AS page_location,
+        event_timestamp
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_*` a
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        _TABLE_SUFFIX
+        BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(@submission_date, INTERVAL 1 DAY))
+        AND FORMAT_DATE('%Y%m%d', @submission_date)
+        AND e.key = 'entrances'
+        AND e.value.int_value = 1
+    ),
+    landing_page_by_session AS (
+      SELECT
+        ga_client_id,
+        ga_session_id,
+        page_location,
+        event_timestamp,
+        ROW_NUMBER() OVER (
+          PARTITION BY
+            ga_client_id,
+            ga_session_id
+          ORDER BY
+            event_timestamp ASC
+        ) AS lp_rnk
+      FROM
+        landing_page_by_session_staging
+      QUALIFY
+        lp_rnk = 1
+    ),
+    install_targets_staging AS (
+      SELECT
+        a.user_pseudo_id AS ga_client_id,
+        CAST(e.value.int_value AS string) AS ga_session_id,
+        event_timestamp,
+        event_name AS install_event_name
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_*` a
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        _table_suffix
+        BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(@submission_date, INTERVAL 1 DAY))
+        AND FORMAT_DATE('%Y%m%d', @submission_date)
+        AND e.key = 'ga_session_id'
+        AND e.value.int_value IS NOT NULL
+        AND a.event_name IN (
+          'product_download'
+        ) --using a list so when Stephanie creates more types, we can add here
+    ),
+    last_install_target AS (
+      SELECT
+        ga_client_id,
+        ga_session_id,
+        event_timestamp,
+        install_event_name,
+        ROW_NUMBER() OVER (
+          PARTITION BY
+            ga_client_id,
+            ga_session_id
+          ORDER BY
+            event_timestamp DESC
+        ) AS install_tgt_rnk
+      FROM
+        install_targets_staging
+      QUALIFY
+        install_tgt_rnk = 1
+    ),
+    all_install_targets AS (
+      SELECT
+        ga_client_id,
+        ga_session_id,
+        ARRAY_AGG(install_event_name) AS all_reported_install_targets
```

Comment:
Should this be `ARRAY_AGG(DISTINCT install_event_name)`?

Reply:
We could - it depends on if you want to know how many times they installed something or not - I thought maybe if you had 3 product downloads it could be useful info? But up to you- if you prefer just unique ones, we could do that also.  In GA4 - the # of downloads isn't limited to 1 per session towards conversion goals, so that's why I initially kept all, even if they do the same multiple times - let me know what you prefer!